src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/Common/TensorPrimitives.IUnaryOneToFourOperator.cs

```diff
@@ -0,0 +1,164 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Runtime.Intrinsics;
+
+#pragma warning disable CS8500 // This takes the address of, gets the size of, or declares a pointer to a managed type
+
+namespace System.Numerics.Tensors
+{
+    public static unsafe partial class TensorPrimitives
+    {
+        /// <summary>Operator that takes one input value and returns a single value.</summary>
+        /// <remarks>The input type must be half the size of the output type.</remarks>
+        private interface IUnaryOneToFourOperator<TInput, TOutput>
+        {
+            static abstract bool Vectorizable { get; }
+            static abstract TOutput Invoke(TInput x);
+            static abstract (Vector128<TOutput>, Vector128<TOutput>, Vector128<TOutput>, Vector128<TOutput>) Invoke(Vector128<TInput> x);
```

Comment:
Just looking at the interface, I'm not sure I understand the connection between the scalar invoke that takes one input and returns one output with the vectorized variants that take one vector of inputs and return four vectors of outputs. Is it perhaps because of the size of `TOutput` being four times as big?

Reply:
Right, it's because you're going from `Nx TInput` to `Nx TOutput`

For scalar, that's just `TInput` to `TOutput`

For `Vector128<TInput>` you can only hold `128 / sizeof(TInput)` elements and if `sizeof(TOutput) == (sizeof(TIntput) * 4)` then you need `4x Vector128<TOutput>` to hold the same element count (i.e. `Vector128<byte>` is 16 bytes, so you need `4x Vector128<float>` since each `Vector128<float>` is 4 floats).